alica_engine/src/engine/modelmanagement/ModelManager.cpp

```diff
@@ -30,69 +30,33 @@
 namespace alica
 {
 
-ModelManager::ModelManager(ConfigChangeListener& configChangeListener, const std::string& domainConfigFolder, PlanRepository& planRepository)
+ModelManager::ModelManager(ConfigChangeListener& configChangeListener, const std::vector<std::string>& domainConfigFolders, PlanRepository& planRepository)
         : _configChangeListener(configChangeListener)
-        , domainConfigFolder(domainConfigFolder)
+        , _domainConfigFolders(domainConfigFolders)
         , _planRepository(planRepository)
 {
-    auto reloadFunctionPtr = std::bind(&ModelManager::reload, this, std::placeholders::_1);
-    _configChangeListener.subscribe(reloadFunctionPtr);
-    reload(_configChangeListener.getConfig());
     Factory::setModelManager(this);
 }
 
-void ModelManager::reload(const YAML::Node& config)
-{
-    basePlanPath = getBasePath("PlanDir");
-    baseRolePath = getBasePath("RoleDir");
-    baseTaskPath = getBasePath("TaskDir");
-}
-
-std::string ModelManager::getBasePath(const std::string& configKey)
-{
-    YAML::Node& config = _configChangeListener.getConfig();
-    std::string basePath;
-    try {
-        basePath = config["Alica"][configKey].as<std::string>();
-    } catch (const std::runtime_error& error) {
-        AlicaEngine::abort(LOGNAME, "Directory for config key '", configKey, "' does not exist in the Alica config file.\n", error.what());
-    }
-
-    if (!essentials::FileSystem::endsWith(basePath, essentials::FileSystem::PATH_SEPARATOR)) {
-        basePath = basePath + essentials::FileSystem::PATH_SEPARATOR;
-    }
-
-    if (!essentials::FileSystem::isPathRooted(basePath)) {
-        basePath = essentials::FileSystem::combinePaths(domainConfigFolder, basePath);
-    }
-
-    if (!essentials::FileSystem::pathExists(basePath)) {
-        AlicaEngine::abort(LOGNAME, "Base path '", basePath, "' does not exist for '" + configKey + "'");
-    }
-
-    Logging::logDebug(LOGNAME) << "Config key '" << configKey << "' maps to '" << basePath << "'";
-    return basePath;
-}
-
 Plan* ModelManager::loadPlanTree(const std::string& masterPlanName)
 {
     std::string masterPlanPath;
-    if (!essentials::FileSystem::findFile(basePlanPath, masterPlanName + alica::Strings::plan_extension, masterPlanPath)) {
+    if (masterPlanPath = findFile(masterPlanName + alica::Strings::plan_extension); masterPlanPath.empty()) {
         AlicaEngine::abort(LOGNAME, "Cannot find MasterPlan '", masterPlanName, "'");
     }
 
-    filesParsed.push_back(masterPlanPath);
+    filesParsed.push_back(masterPlanName + alica::Strings::plan_extension);
     Plan* masterPlan = (Plan*) parseFile(masterPlanPath, alica::Strings::plan);
     while (filesToParse.size() > 0) {
-        std::string fileToParse = filesToParse.front();
-        filesToParse.pop_front();
+        std::string fileToParse;
+        if (fileToParse = findFile(filesToParse.front()); fileToParse.empty()) {
```

Comment:
technically you can initialize variable on creation
```suggestion
        const auto fileToParse = findFile(filesToParse.front());
        if (fileToParse.empty()) {
```